src/commands/Minion/alch.ts

```diff
@@ -26,11 +26,94 @@ const unlimitedFireRuneProviders = resolveItems([
 	'Tome of fire'
 ]);
 
+export function calculateAlchItemByTripDuration(params: {
+	user: KlasaUser;
+	duration: number;
+	items?: { item: Item; qty: number }[];
+	qtyLimit?: number;
+	calculateByRunesOwned?: boolean;
+}) {
+	let { items, duration, user, qtyLimit, calculateByRunesOwned } = params;
+	const userBank = user.bank().clone();
+
+	if (!items) items = [];
+
+	if (items.length === 0) {
+		for (const i of user.getUserFavAlchs() as Item[]) {
+			// Ignore alchs that are already on the alch items
+			if (items.find(a => a.item.id === i.id)) continue;
+			items.push({
+				item: i,
+				qty: qtyLimit
+					? qtyLimit < userBank.amount(i.id)
+						? qtyLimit
+						: userBank.amount(i.id)
+					: userBank.amount(i.id)
+			});
+		}
+		if (items.length === 0) return false;
+	}
+
+	// Check if user has infinite fire runes
+	let infiniteFireRunes = false;
+	for (const runeProvider of unlimitedFireRuneProviders) {
+		if (user.hasItemEquippedAnywhere(runeProvider)) {
+			infiniteFireRunes = true;
+			break;
+		}
+	}
+
+	// 5 tick action
+	const timePerAlch = Time.Second * 3;
+	let maxCasts = Math.floor(duration / timePerAlch);
+	if (qtyLimit && maxCasts > qtyLimit) maxCasts = qtyLimit;
+
+	// If calculateByRunesOwned is true, limit casts to what the user have runes for
+	if (calculateByRunesOwned) {
+		const maxRuneCasts = userBank.fits(
+			new Bank().add('Fire rune', infiniteFireRunes ? 0 : 5).add('Nature rune', 1)
+		);
+		if (maxCasts > maxRuneCasts) {
+			maxCasts = maxRuneCasts;
+		}
+	}
+
+	const alchBank = new Bank();
+	let totalValue = 0;
+
+	let castsLeft = maxCasts;
+	for (const i of items) {
```

Comment:
Should probably add some sorting so the highest alch value items are alch'd first.